### PR TITLE
add shellcheck as a check-in workflow

### PR DESCRIPTION
**Pull request by eriksywu**

### CODE REVIEW

File: .pipelines/scripts/verify_shell.sh [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        echo "distro ${DISTRO} not supported at this time. skipping shellcheck"
+        return
+    fi
+fi
+
+filesToCheck=$(find . -type f -name "*.sh" -not -path './parts/linux/cloud-init/artifacts/*')
+
+echo "Running shellcheck..."
+
+IGNORED="
```
Comment (eriksywu) [inline .pipelines/scripts/verify_shell.sh:22]: need to figure out which errors/warning to ignore

File: .pipelines/scripts/verify_shell.sh [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        apt-get install shellcheck
+    elif [[ "${DISTRO}" == "darwin" ]]; then
+        brew install cabal-install
+    else 
+        echo "distro ${DISTRO} not supported at this time. skipping shellcheck"
+        return
+    fi
+fi
+
+filesToCheck=$(find . -type f -name "*.sh" -not -path './parts/linux/cloud-init/artifacts/*')
```
Comment (xuto2) [inline .pipelines/scripts/verify_shell.sh:18]: should also exclude ./pkg/agent/testdata/

File: .pipelines/scripts/verify_shell.sh (hunk 1 above)
Comment (xuto2) [reply .pipelines/scripts/verify_shell.sh:22]: have you tried those in aks-engine? https://github.com/Azure/aks-engine/blob/f2a577d0ab550b769d5abaf276e45e8e6f95bfdd/scripts/validate-shell.sh#L17

Comment (eriksywu) [reply .pipelines/scripts/verify_shell.sh:22]: ah yea of course...i'll get those in :)

**Pull request merged**